internal/cmd/opensearch/instance/create/create.go

```diff
@@ -0,0 +1,253 @@
+	cmd.Flags().String(metricsPrefixFlag, "", "Metrics prefix")
+	cmd.Flags().Var(flags.UUIDFlag(), monitoringInstanceIdFlag, "Monitoring instance ID")
+	cmd.Flags().StringSlice(pluginFlag, []string{}, "Plugin")
+	cmd.Flags().Var(flags.CIDRSliceFlag(), sgwAclFlag, "List of IP networks in CIDR notation which are allowed to access this instance")
+	cmd.Flags().StringSlice(syslogFlag, []string{}, "Syslog")
+	cmd.Flags().Var(flags.UUIDFlag(), planIdFlag, "Plan ID")
+	cmd.Flags().String(planNameFlag, "", "Plan name")
+	cmd.Flags().String(versionFlag, "", "Instance OpenSearch version")
+
+	err := flags.MarkFlagsRequired(cmd, instanceNameFlag)
+	cobra.CheckErr(err)
+}
+
+func parseInput(cmd *cobra.Command, service, operation string) (*inputModel, error) {
+	globalFlags := globalflags.Parse(cmd)
+	if globalFlags.ProjectId == "" {
+		return nil, &cliErr.ProjectIdError{}
+	}
+
+	planId := flags.FlagToStringPointer(cmd, planIdFlag)
+	planName := flags.FlagToStringValue(cmd, planNameFlag)
+	version := flags.FlagToStringValue(cmd, versionFlag)
+
+	if planId == nil && (planName == "" || version == "") {
+		return nil, &cliErr.DSAInputPlanError{
+			Service:   service,
+			Operation: operation,
+		}
+	}
+	if planId != nil && (planName != "" || version != "") {
+		return nil, &cliErr.DSAInputPlanError{
+			Service:   service,
+			Operation: operation,
+		}
+	}
+
+	return &inputModel{
+		GlobalFlagModel:      globalFlags,
+		InstanceName:         flags.FlagToStringPointer(cmd, instanceNameFlag),
+		EnableMonitoring:     flags.FlagToBoolPointer(cmd, enableMonitoringFlag),
+		MonitoringInstanceId: flags.FlagToStringPointer(cmd, monitoringInstanceIdFlag),
+		Graphite:             flags.FlagToStringPointer(cmd, graphiteFlag),
+		MetricsFrequency:     flags.FlagToInt64Pointer(cmd, metricsFrequencyFlag),
+		MetricsPrefix:        flags.FlagToStringPointer(cmd, metricsPrefixFlag),
+		Plugin:               flags.FlagToStringSlicePointer(cmd, pluginFlag),
+		SgwAcl:               flags.FlagToStringSlicePointer(cmd, sgwAclFlag),
+		Syslog:               flags.FlagToStringSlicePointer(cmd, syslogFlag),
+		PlanId:               planId,
+		PlanName:             planName,
+		Version:              version,
+	}, nil
+}
+
+type OpenSearchClient interface {
+	CreateInstance(ctx context.Context, projectId string) opensearch.ApiCreateInstanceRequest
+	ListOfferingsExecute(ctx context.Context, projectId string) (*opensearch.ListOfferingsResponse, error)
+}
+
+func buildRequest(ctx context.Context, service string, model *inputModel, apiClient OpenSearchClient) (opensearch.ApiCreateInstanceRequest, error) {
+	req := apiClient.CreateInstance(ctx, model.ProjectId)
+
+	var planId *string
+	var err error
+
+	offerings, err := apiClient.ListOfferingsExecute(ctx, model.ProjectId)
+	if err != nil {
+		return req, fmt.Errorf("get OpenSearch offerings: %w", err)
+	}
+
+	if model.PlanId == nil {
+		planId, err = opensearchUtils.LoadPlanId(service, model.PlanName, model.Version, offerings)
```

Comment:
Same comment as `parseInput`'s signature: `service` looks unnecessary in `LoadPlanId`'s signature
Here it's particularly weird, because it's coming from a package named `opensearchUtils`, it really shouldn't need to be told what service this is